### ISSUE TITLE
[Menu] Shortcut and Combination variants require "aria-checked" on all selectable menu items

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3006]
### Provide a general summary of the issue here

In [Menu Shortcut](https://storybook.sanomalearning.design/?path=/story/actions-menu-menu--shortcut) and [Menu Combination](https://storybook.sanomalearning.design/?path=/story/actions-menu-menu--combination) not selected `selectable` menu items do not have `aria-checked='false'` added. Only selected menu item has `aria-checked='true'` which is not sufficient for assistive technologies.
All selectable menu items should have `aria-checked` at all times. 
Not selected menu items should have `aria-checked='false'` atribute.

<img width="178" height="141" alt="Screenshot Menu Shortcus" src="https://github.com/user-attachments/assets/89114cdf-3ae1-4abb-a482-6b97709a2b6a" />
<img width="199" height="309" alt="Screenshot Menu Combination" src="https://github.com/user-attachments/assets/56d8d61b-c920-4df6-a661-f56e7abe3514" />

### 🤔 Expected Behavior?

Checked `<sl-menu-item selectable selected>` should have `aria-chechked='true'`.
Non-checked `<sl-menu-item selectable>` should have `aria-chechked='false'` at all times.
It only applies to `selectable` menu items.

### 😯 Current Behavior

VoiceOver is announcing not checked menu items as "NAME, menu item" which disturb information about relationship of menu items. 
`axe-core` automated tests report issue with "ARIA required attributes".

### 🖥️ Steps to Reproduce

1. Check if `<sl-menu-item>` has `selectable` attibute.
2. If it is selectable and is currently selected it should have `aria-checked='true'`
3. If it is selectable but is not currently selected it should have `aria-checked='false'`
4. On selected menu item NVDA should announce "NAME, checked", VoiceOver should announce "NAME, checked, menu item"
5. On not selected menu item NVDA should announce "NAME, not checked", VoiceOver should announce "NAME, not checked, menu item"

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

macOS and Win11

### 👤 Your name

SLDS team

### 🧢 Your product/team

SLDS team

### 🎨 Your Theme(s)

Sanoma Learning
